File: lib/models/livraison.dart
class Livraison {
  final int? id;
  final int commandeId;
  final String adresseLivraison;
  final String statut;

  Livraison({
    this.id,
    required this.commandeId,
    required this.adresseLivraison,
    required this.statut,
  });

  factory Livraison.fromJson(Map<String, dynamic> json) => Livraison(
        id: json['id'],
        commandeId: json['commande']['id'],
        adresseLivraison: json['adresseLivraison'],
        statut: json['statut'],
      );

  Map<String, dynamic> toJson() => {
        'commandeId': commandeId,
        'adresseLivraison': adresseLivraison,
        'statut': statut,
      };
}

File: src/main/java/com/ouertatani/gestionproduit/model/Livraison.java
package com.ouertatani.gestionproduit.model;

import lombok.*;
import javax.persistence.*;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Livraison {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "commande_id", nullable = false)
    private Commande commande;

    @Lob
    private String adresseLivraison;

    @Enumerated(EnumType.STRING)
    private StatutLivraison statut = StatutLivraison.EN_ATTENTE;

    public enum StatutLivraison {
        EN_ATTENTE, EN_COURS, LIVREE, ANNULEE
    }
}
